web/src/features/map-controls/DarkModeSelector.tsx

```diff
@@ -0,0 +1,77 @@
+import { ReactElement, useState } from 'react';
+import { BsMoonStars } from 'react-icons/bs';
+import { HiOutlineComputerDesktop, HiOutlineSun } from 'react-icons/hi2';
+import { twMerge } from 'tailwind-merge';
+import { useTranslation } from 'translation/translation';
+
+interface DarkModeSelectorProperties {
+  setDarkModeSelectorOpen: (isOpen: boolean) => void;
+  className?: string;
+}
+
+function getIcon(key: string) {
+  switch (key) {
+    case 'light': {
+      return <HiOutlineSun size={20} />;
+    }
+    case 'dark': {
+      return (
+        <BsMoonStars
+          size={14}
+          style={{ strokeWidth: '0.2', marginLeft: 3, marginRight: 2 }}
+        />
+      );
+    }
+    case 'system': {
+      return <HiOutlineComputerDesktop size={18} />;
+    }
+    default: {
+      return null;
+    }
+  }
+}
+
+export default function DarkModeSelector({
+  setDarkModeSelectorOpen,
+  className,
+}: DarkModeSelectorProperties): ReactElement {
+  const { __, i18n } = useTranslation();
+  const darkModeKeys: Array<string> = ['light', 'dark', 'system'];
+  const [selectedThemeMode, setSelectedThemeMode] = useState(
+    localStorage.getItem('theme') || 'system'
```

Comment:
I think in this case || is better let's say you do localstorage.setItem("theme", "") to remove darkmode

localStorage.getItem('theme') ?? system; //Output: ''
localStorage.getItem('theme') ?? system; //Output: 'system'

Reply:
But the state should never be an empty string?
At first visit it will be null and then use system.
Then if a user changes it the value will be set to either dark, light or system.
If the user then change it from light to dark the value will still be there but just change.

I just can't see any potential scenario where it would be a empty string unless someone goes in an manually changes the local storage but I don't think that is something we should account for.

The flow would be something like this:
First visit when localStorage theme is null/undefined.
`localStorage.getItem('theme') ?? system` results in the system theme being used.

User changes it to dark:
localstorage.setItem("theme", "dark") is invoked and the theme key is now dark.

User changes it to light:
localstorage.setItem("theme", "light") is invoked and the theme key is now light.

User cleans cache and local storage and the key is now null again and the fallback will be used like in the first step.

So unless we set it to a empty string somewhere in the code it should always be either undefined/null, light, dark or system?

Or have I missed something?

Reply:
Someone might change the implementation in the future, and it's possible that a future implementation could set the theme key to an empty string. It's not a good implementation but it's a common pattern where people use setItem('x','') instead of removeItem('x')

I can see why (albeit a small edge case) || is better than ?? but I can't see why ?? would be better than ||.

Reply:
I don't think you can apply a hard rule that every || should be a ?? :)

Reply:
> Someone might change the implementation in the future, and it's possible that a future implementation could set the theme key to an empty string. It's not a good implementation but it's a common pattern where people use setItem('x','') instead of removeItem('x')
> 
> I can see why (albeit a small edge case) || is better than ?? but I can't see why ?? would be better than ||.

I suppose that is true and it technically don't hurt to keep the || but I do think we should use strict typing with type literals, possibly use enums and follow "best practices" and never set it to an empty string.

There is a argument to be made that ?? is better for type safety but I suppose that don't really apply to this case in the way we have implemented it.

Reply:
Maybe I'm biased after years of living in a type free world 🤠